deps/v8/src/snapshot/deserializer.cc

```diff
@@ -151,6 +153,9 @@ MaybeHandle<Object> Deserializer::DeserializePartial(
   // changed and logging should be added to notify the profiler et al of the
   // new code, which also has to be flushed from instruction cache.
   CHECK_EQ(start_address, code_space->top());
+
+  if (FLAG_rehash_snapshot && can_rehash_) RehashContext(Context::cast(root));
```

Comment:
For my own curiosity, `root` is logically always a Context here?  I followed the logic down to Deserializer::ReadData() but things got complex fast from there on.